File: src/main/java/cz/solutions/cockroach/DividendReport.java
package cz.solutions.cockroach;

import lombok.Value;

import java.util.List;
import java.util.Map;

@Value
public class DividendReport {
    List<PrintableDividend> printableDividendList;
    double totalBruttoDollar;
    double totalTaxDollar;
    double totalBruttoCrown;
    double totalTaxCrown;
    double totalTaxReversalDollar;
    double totalTaxReversalCrown;

    public Map<String,?> asMap(){
        return Map.of(
                "dividendList", printableDividendList,
                "totalBruttoDollar", FormatingHelper.formatDouble(totalBruttoDollar),
                "totalTaxDollar", FormatingHelper.formatDouble(totalTaxDollar),
                "totalBruttoCrown", FormatingHelper.formatDouble(totalBruttoCrown),
                "totalTaxCrown", FormatingHelper.formatDouble(totalTaxCrown),
                "totalTaxReversal", totalTaxReversalDollar > 0 ? FormatingHelper.formatDouble(totalTaxReversalDollar) : "",
                "totalTaxReversalCrown", totalTaxReversalCrown > 0 ? FormatingHelper.formatDouble(totalTaxReversalCrown) : ""
        );
    }
}

File: src/main/kotlin/cz/solutions/cockroach/DividendReport.kt
package cz.solutions.cockroach

data class CurrencyDividendSection(
    val currency: Currency,
    val printableDividendList: List<PrintableDividend>,
    val totalBrutto: Double,
    val totalTax: Double,
    val totalBruttoCrown: Double,
    val totalTaxCrown: Double,
    val totalTaxReversal: Double,
    val totalTaxReversalCrown: Double
)

data class CzkDividendSection(
    val printableDividendList: List<PrintableCzkDividend>,
    val totalBruttoCrown: Double,
    val totalTaxCrown: Double,
    val totalTaxReversalCrown: Double
)

data class DividendReport(
    val sections: List<CurrencyDividendSection>,
    val czkSection: CzkDividendSection?
) {
    val totalNonCzkBruttoCrown: Double get() = sections.sumOf { it.totalBruttoCrown }
    val totalNonCzkTaxCrown: Double get() = sections.sumOf { it.totalTaxCrown }
}